node-notes-cli/app.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-console */
+const fs = require('fs');
+const json = require('./data.json');
+let count = 0;
+
+if (process.argv[2] === 'read') {
+  console.log(JSON.stringify(json.notes, null, 2));
+} else if (process.argv[2] === 'create') {
+  const arr = [];
+  for (const key in json.notes) {
+    count++;
+    arr.push(Number(key));
+  }
+  count = Math.max(...arr); // this allows 'create' to function properly after the user used 'delete' on an entry.
+  json.notes[count + 1] = process.argv[3];
```

Comment:
You can simplify your code a bit by just incrementing the `data.nextId` instead of keeping a separate counter.